hazelcast/src/main/java/com/hazelcast/jet/JobListener.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet;
+
+import java.util.UUID;
+
+/**
+ * @see Job#addStatusListener(JobListener)
+ * @see Job#removeStatusListener(UUID)
+ *
+ * @since 5.3
+ */
+public interface JobListener {
+    void jobStatusChanged(JobEvent event);
+}
```

Comment:
This is public API, it needs rigorous javadoc.

Also is this JobListener listening for all events on the Job (which would the name `JobListener` suggest, or is it a listener listening for job status changes, as the name of the method `jobStatusChanged` and `Job#addStatusListener` say?

I would rename this in the following way:

- `JobListener` -> `JobStatusListener`
- `JobEvent` -> `JobStatusEvent`
- `Job#addStatusListener` -> `addJobStatusListener` (here it's a bit redundant, but I would still do it)